Gems/RecastNavigation/Code/Source/Components/RecastNavigationMeshComponent.cpp

```diff
@@ -65,40 +71,58 @@ namespace RecastNavigation
             &RecastNavigationProviderRequests::CollectGeometry,
             m_meshConfig.m_tileSize, aznumeric_cast<float>(m_meshConfig.m_borderSize) * m_meshConfig.m_cellSize);
 
-        for (AZStd::shared_ptr<TileGeometry>& tile : tiles)
         {
-            if (tile->IsEmpty())
-            {
-                continue;
-            }
-
-            // Given geometry create Recast tile structure.
-            NavigationTileData navigationTileData = CreateNavigationTile(tile.get(),
-                m_meshConfig, m_context.get());
-
-            // If a tile at the location already exists, remove it before update the data.
-            if (const dtTileRef tileRef = m_navObject->m_mesh->getTileRefAt(tile->m_tileX, tile->m_tileY, 0))
-            {
-                m_navObject->m_mesh->removeTile(tileRef, nullptr, nullptr);
-            }
+            NavMeshQuery::LockGuard lock = GetNavigationObject()->AcquireLock();
```

Comment:
Rather than risk weird behaviour without RVO, why not just write:

`NavMeshQuery::LockGuard lock(GetNavigationObject());`?

Reply:
That is actually what I had a day ago in my work in progress branch for this PR. I have been debating what looks prettier and cleaner. But yeah, I think this is safer than relying on invisible RTO.

Reply:
Done.